File: src/01-easy/retro-typewriter-art/solution.vb
Module Solution
    Sub Main()
        Dim line As String = Console.ReadLine()
        Dim sb As New System.Text.StringBuilder()
        For Each tok As String In line.Split(" "c)
            If tok.Length = 0 Then Continue For
            Dim l2 As String = If(tok.Length >= 2, tok.Substring(tok.Length - 2), "")
            Dim ch As String
            Dim num As String
            If l2 = "nl" Then
                ch = vbLf : num = tok.Substring(0, tok.Length - 2)
            ElseIf l2 = "sp" Then
                ch = " " : num = tok.Substring(0, tok.Length - 2)
            ElseIf l2 = "bS" Then
                ch = "\" : num = tok.Substring(0, tok.Length - 2)
            ElseIf l2 = "sQ" Then
                ch = "'" : num = tok.Substring(0, tok.Length - 2)
            Else
                ch = tok.Substring(tok.Length - 1) : num = tok.Substring(0, tok.Length - 1)
            End If
            Dim n As Integer = If(num.Length = 0, 1, Math.Max(1, CInt(num)))
            For i = 1 To n : sb.Append(ch) : Next
        Next
        Console.WriteLine(sb.ToString())
    End Sub
End Module

File: src/01-easy/retro-typewriter-art/solution.cs
using System;
using System.Collections.Generic;
class Solution {
    static void Main(){
        var ab=new Dictionary<string,char>{{"sp",' '},{"bS",'\\'},{"sQ",'\''},{"nl",'\n'}};
        var line=Console.ReadLine();
        var sb=new System.Text.StringBuilder();
        foreach(var tok in line.Split(' ')){
            char ch; string num;
            var l2=tok.Length>=2?tok.Substring(tok.Length-2):"";
            if(ab.ContainsKey(l2)){ch=ab[l2];num=tok.Substring(0,tok.Length-2);}
            else{ch=tok[tok.Length-1];num=tok.Substring(0,tok.Length-1);}
            int n=num.Length>0?int.Parse(num):1;
            sb.Append(new string(ch,Math.Max(1,n)));
        }
        Console.WriteLine(sb);
    }
}
